### PR TITLE
feat: remove --verbose from pebble if base is 24.04 or later

### PR DESCRIPTION
**Pull request by linostar**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

This is a subtask of https://github.com/canonical/rockcraft/pull/495.
We decided to remove `--verbose` from base 24.04 for starters, and see the impact.

### CODE REVIEW

**Comment by linostar** [comment]
@cjdcordeiro @tigarmo Ready for review.

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
I think this deserves a spread test

**Comment by linostar** [comment]
> I think this deserves a spread test

On the way, Chef.

**Comment by linostar** [comment]
@tigarmo There is the odd case of `base: ubuntu@22.04` with `build-base: devel`. It is unlikely to be used by anyone, but do you want it to be handled? and if yes, handled like 22.04 or 24.04?

**Comment by linostar** [comment]
> > @tigarmo There is the odd case of `base: ubuntu@22.04` with `build-base: devel`. It is unlikely to be used by anyone, but do you want it to be handled? and if yes, handled like 22.04 or 24.04?
> 
> If you have a `build-base` you can always rely on it. For your specific example, the semantics are: use whatever is the current `devel` buildd image for the building, and `ubuntu@22.04` for the target/base layer of the final rock. So it should have the _new_ behavior.

That's what I currently opted to do. Rely on build-base if it exists, otherwise falls back to base (pun intended).

**Review by tigarmo — CHANGES_REQUESTED** [review]

File: rockcraft/oci.py [hunk 1]
```diff
@@ -353,11 +353,16 @@ def set_default_user(self, user: str) -> None:
         _config_image(image_path, params)
         emit.progress(f"Default user set to {user}")
 
-    def set_entrypoint(self, entrypoint_service: str | None = None) -> None:
+    def set_entrypoint(
+        self, entrypoint_service: str | None = None, base: str = "ubuntu@22.04"
```
[the remark this answers is not in the document]
Comment (linostar) [reply rockcraft/oci.py]: Done

File: tests/spread/rockcraft/non-verbose/task.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  run_rockcraft pack
+
+  test -f non-verbose-test_latest_amd64.rock
+
+  # test container execution
+  sudo /snap/rockcraft/current/bin/skopeo --insecure-policy copy oci-archive:non-verbose-test_latest_amd64.rock docker-daemon:non-verbose-test:latest
+  rm non-verbose-test_latest*.rock  
+  docker images non-verbose-test:latest
+  id=$(docker run -d non-verbose-test)
+  test "$(docker inspect "$id" -f '{{json .Config.Entrypoint}}')" = '["/bin/pebble","enter"]'
```
[the remark this answers is not in the document]
Comment (linostar) [reply tests/spread/rockcraft/non-verbose/task.yaml]: Done

**Comment by linostar** [comment]
@tigarmo This should be ready for merge. The last spread test took more than 2 hours smh and failed, but the related spread tests to this PR were passing. I merged main branch into this to restart the checks.

**Review by tigarmo — APPROVED** [review]
thanks!

File: rockcraft/oci.py [hunk 3]
```diff
@@ -353,11 +353,14 @@ def set_default_user(self, user: str) -> None:
         _config_image(image_path, params)
         emit.progress(f"Default user set to {user}")
 
-    def set_entrypoint(self, entrypoint_service: str | None = None) -> None:
+    def set_entrypoint(self, entrypoint_service: str | None, build_base: str) -> None:
         """Set the OCI image entrypoint. It is always Pebble."""
         emit.progress("Configuring entrypoint...")
         image_path = self.path / self.image_name
-        entrypoint = [f"/{Pebble.PEBBLE_BINARY_PATH}", "enter", "--verbose"]
+        if build_base in ["ubuntu@20.04", "ubuntu@22.04"]:
```
Comment (SamirPS) [inline rockcraft/oci.py:360]: Can you just put the list without verbose by default and just append the version of base in 20.04 or 22.04?

File: tests/unit/test_oci.py [hunk 4]
```diff
@@ -585,9 +585,31 @@ def test_set_entrypoint_default(self, mock_run):
             ),
         ]
 
+    def test_set_entrypoint_default_no_verbose(self, mock_run):
```
Comment (SamirPS) [inline tests/unit/test_oci.py:588]: The two test are quite the same, why not using this : pytest.mark.parametrize

**Review by SamirPS — CHANGES_REQUESTED** [review]

File: rockcraft/oci.py (hunk 3 above)
Comment (SamirPS) [reply rockcraft/oci.py:360]: This will permit to remove the else and be more cleaner

File: tests/unit/test_oci.py (hunk 4 above)
Comment (linostar) [reply tests/unit/test_oci.py]: Thanks for the suggestion! Done.

File: rockcraft/oci.py (hunk 3 above)
Comment (linostar) [reply rockcraft/oci.py:361]: Done

**Review by SamirPS — APPROVED** [review]

**Comment by tigarmo** [comment]
@linostar @cjdcordeiro there are some new failures on chisel + 24.04 now. Unrelated to this PR but we need to look into this:

```
 2024-04-08 07:37:57.090 :: 2024-04-08 07:37:56.518 :: 2024/04/08 07:37:56 Release date: Mon, 08 Apr 2024  6:56:37 UTC
2024-04-08 07:37:57.090 :: 2024-04-08 07:37:56.519 :: 2024/04/08 07:37:56 Fetching index for ubuntu 24.04 noble-updates main component...
2024-04-08 07:37:57.090 :: 2024-04-08 07:37:56.551 :: 2024/04/08 07:37:56 Fetching index for ubuntu 24.04 noble-updates universe component...
2024-04-08 07:37:57.090 :: 2024-04-08 07:37:56.633 :: error: slice package "liblttng-ust-common1" missing from archive
2024-04-08 07:37:57.090 :: 2024-04-08 07:37:56.643 Failed to cut requested chisel slices: dotnet-runtime-8.0_libs
```

**Comment by linostar** [comment]
@tigarmo packages.ubuntu.com seems to be partially (?) down.
https://packages.ubuntu.com/noble/ is giving HTTP 500.

**Review by cjdcordeiro — APPROVED** [review]
LGTM thanks

**Pull request merged**